spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/MessageListenerAdapter.java

```diff
@@ -339,11 +339,33 @@ protected String getListenerMethodName(Message originalMessage, Object extracted
 	 * @param extractedMessage the content of the message
 	 * @return the array of arguments to be passed into the listener method (each element of the array corresponding to
 	 * a distinct method argument)
+	 * @deprecated use @{org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter#buildListenerArguments
+	 * (java.lang.Object, com.rabbitmq.client.Channel, org.springframework.amqp.core.Message)} to get complete arguments
 	 */
+	@SuppressWarnings("deprecation")
```

Comment:
should be on the new method

Reply:
does old method need any @Deprecated or Supperwarning?

Reply:
and here `@Deprecated` as an annotation

Reply:
ok